src/transformers/pipelines.py

```diff
@@ -469,6 +479,19 @@ def ensure_tensor_on_device(self, **inputs):
         """
         return {name: tensor.to(self.device) for name, tensor in inputs.items()}
 
+    def check_model_type(self, supported_models):
+        """
+        Check if the model class is in the supported class list of the pipeline.
+        """
+        if not isinstance(supported_models, list):  # Create from a model mapping
+            supported_models = [item[1].__name__ for item in supported_models.items()]
+        if self.model.__class__.__name__ not in supported_models:
+            raise PipelineException(
```

Comment:
I would warn rather than raise an error, because raising an exception here requires us to keep `supported_models` completely up to date.

Reply:
For all pipelines except `TextGenerationPipeline`, `supported_models` are automatically constructed from `AutoModel` which is certainly up-to-date. Plus, `pipeline` is a "production" style API, using a newly added model seems to be a corner case.

Reply:
Also, I throw errors following the original logic of `SummarizationPipeline`, which maintains an `ALLOW_LIST` and throws an error if the model is not on that list.